### PR TITLE
Announcements not showing

### PR DESCRIPTION
**Pull request by akench**
Refactored view model to observe directly on changes from database
Moved upsert operation to `BaseDao`
Added announcement body inside scrollview for smooth scrolling
Added a `FloatingActionButton` to scroll to the top
Saves the position of the announcement you are looking at when leaving the fragment, and restores it

### CODE REVIEW

File: app/src/main/java/com/sakaimobile/development/sakaiclient20/persistence/SakaiDatabase.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
         Attachment.class,
         Announcement.class,
         Resource.class
-}, version = 9)
+}, version = 13)
```
Comment (SChakravorti21) [inline app/src/main/java/com/sakaimobile/development/sakaiclient20/persistence/SakaiDatabase.java]: Can we leave the version at 9 for now? I want to avoid incrementing the version unless we make schema changes for a major release, so the version number should only increment by 1 iff. we have a major release coming up. If you clear app data we won't get the runtime exception about schema changing.

File: app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/adapters/AnnouncementsAdapter.java [hunk 2]
```diff
@@ -126,100 +106,58 @@ public void onClick(View v) {
...
-                }
+            //if site announcements, show announcement title, then announcement body
+            holder.cardHeading2.setText(currAnnouncement.title);
 
+            try {
+                //if its after android N, use this method for setting the html
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                    holder.cardHeading3.setText(Html.fromHtml(currAnnouncement.body, Html.FROM_HTML_MODE_COMPACT));
+                } else {
+                    holder.cardHeading3.setText(Html.fromHtml(currAnnouncement.body));
```
Comment (SChakravorti21) [inline app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/adapters/AnnouncementsAdapter.java]: What are `cardHeading1`, `cardHeading2`, and `cardHeading3`? This is a bit hard to follow, perhaps rename them to be more representative of the information shown in the views.

File: app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/fragments/AnnouncementsFragment.java [hunk 3]
```diff
@@ -113,10 +110,14 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
         announcementRecycler.setLayoutManager(layoutManager);
         announcementRecycler.setItemAnimator(new DefaultItemAnimator());
 
+        announcementRecycler.setItemViewCacheSize(20);
```
Comment (SChakravorti21) [inline app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/fragments/AnnouncementsFragment.java]: I assume this is needed because fast scrolling would drop frames if enough views aren't cached, so it would be nice if you could add a comment about that here (in case we ever come back to this). Or, if that's not the case, then explain what the large view cache is needed for.

File: app/src/main/res/layout/fragment_full_announcement.xml [hunk 4]
```diff
@@ -91,12 +91,20 @@
...
-                    android:textSize="15sp" />
+                    android:layout_height="match_parent">
+
+                    <TextView
+                        android:id="@+id/announcement_content"
+                        android:layout_width="match_parent"
+                        android:layout_height="match_parent"
+                        android:text="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque mollis, lectus id dictum blandit, elit elit auctor metus, in ultrices velit justo in odio. Maecenas metus nibh, elementum quis orci sit amet, sodales tincidunt est. Nam mollis aliquet lorem ac sodales. Pellentesque posuere convallis tellus, vitae dignissim felis fermentum sit.Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque mollis, lectus id dictum blandit, elit elit auctor metus, in ultrices velit justo in odio. Maecenas metus nibh, elementum quis orci sit amet, sodales tincidunt est. Nam mollis aliquet lorem ac sodales. Pellentesque posuere convallis tellus, vitae dignissim felis fermentum sit."
```
Comment (SChakravorti21) [inline app/src/main/res/layout/fragment_full_announcement.xml]: If you're using the `text` attribute to preview what the view looks like in Android Studio, then always use `tools:text` instead. `android:text` actually sets the `TextView` text at runtime, which is a waste of resources.

File: app/src/main/res/layout/fragment_full_announcement.xml [hunk 5]
```diff
@@ -91,12 +91,20 @@
                 android:paddingTop="10dp"
                 >
 
-                <TextView
-                    android:id="@+id/announcement_content"
+                <ScrollView
```
Comment (SChakravorti21) [inline app/src/main/res/layout/fragment_full_announcement.xml]: What is the `ScrollView` needed for inside the card?

Comment (akench) [reply app/src/main/res/layout/fragment_full_announcement.xml]: it makes scrolling in the announcement body much smoother (ie. u can flick up and down to scroll)

Comment (SChakravorti21) [reply app/src/main/res/layout/fragment_full_announcement.xml]: Oh shoot I didn't notice this is the full card, my bad, this isn't an issue lol.
